File: step_definitions/Successful Integration and Transaction Using Stripe.js
import { Given, When, Then } from "cypress-cucumber-preprocessor/steps";
import PaymentPage from "../pages/paymentPage";

Given("Stripe account setup is completed", () => {
 PaymentPage.verifyStripeSetup();
});

When("I navigate to the payment page", () => {
 PaymentPage.visit();
});

Then("the payment page is displayed correctly", () => {
 PaymentPage.verifyPageDisplayed();
});

Then("API keys are configured", () => {
 PaymentPage.verifyAPIKeysConfigured();
});

When("I select Stripe as the payment method", () => {
 PaymentPage.selectStripePaymentMethod();
});

Then("Stripe payment method is selected", () => {
 PaymentPage.verifyStripePaymentMethodSelected();
});

When("I enter valid credit card details", () => {
 PaymentPage.enterCreditCardDetails();
});

Then("credit card details are accepted", () => {
 PaymentPage.verifyCreditCardDetailsAccepted();
});

When("I enter valid billing address", () => {
 PaymentPage.enterBillingAddress();
});

Then("billing address is accepted", () => {
 PaymentPage.verifyBillingAddressAccepted();
});

When("I enter valid email address", () => {
 PaymentPage.enterEmailAddress();
});

Then("email address is accepted", () => {
 PaymentPage.verifyEmailAddressAccepted();
});

When("I enter product ID and amount to be charged", () => {
 PaymentPage.enterProductIdAndAmount();
});

Then("product ID and amount are accepted", () => {
 PaymentPage.verifyProductIdAndAmountAccepted();
});

When("I submit the payment", () => {
 PaymentPage.submitPayment();
});

Then("payment is processed successfully", () => {
 PaymentPage.verifyPaymentProcessed();
});

When("I verify transaction confirmation message", () => {
 PaymentPage.verifyTransactionConfirmationMessage();
});

Then("transaction confirmation message is displayed", () => {
 PaymentPage.verifyTransactionMessageDisplayed();
});

When("I check Stripe dashboard for transaction record", () => {
 PaymentPage.checkStripeDashboard();
});

Then("transaction record is present in Stripe dashboard", () => {
 PaymentPage.verifyTransactionRecordInDashboard();
});

When("I verify email notification for transaction", () => {
 PaymentPage.verifyEmailNotification();
});

Then("email notification is received", () => {
 PaymentPage.verifyEmailNotificationReceived();
});

When("I log out from the application", () => {
 PaymentPage.logout();
});

Then("user is logged out successfully", () => {
 PaymentPage.verifyLogoutSuccess();
});

When("I log back into the application", () => {
 PaymentPage.login();
});

Then("user is logged in successfully", () => {
 PaymentPage.verifyLoginSuccess();
});

When("I check order history for transaction", () => {
 PaymentPage.checkOrderHistory();
});

Then("transaction is present in order history", () => {
 PaymentPage.verifyTransactionInOrderHistory();
});

When("I verify customer support is available for transaction queries", () => {
 PaymentPage.verifyCustomerSupport();
});

Then("customer support is available", () => {
 PaymentPage.verifyCustomerSupportAvailable();
});

When("I ensure transaction details are logged for auditing", () => {
 PaymentPage.ensureTransactionDetailsLogged();
});

Then("transaction details are logged", () => {
 PaymentPage.verifyTransactionDetailsLogged();
});

File: step_definitions/Successful Integration and Transaction Using Stripe.java
package com.cucumber.steps;

import com.page_objects.StripeIntegrationPage;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

public class StripeIntegrationSteps extends StripeIntegrationPage {

    @Given("Stripe account setup is completed")
    public void stripeAccountSetupIsCompleted() {
        verifyStripeAccountSetup();
    }

    @When("I navigate to the payment page")
    public void iNavigateToThePaymentPage() {
        navigateToPaymentPage();
    }

    @Then("the payment page is displayed correctly")
    public void thePaymentPageIsDisplayedCorrectly() {
        verifyPaymentPageDisplayed();
    }

    @Then("API keys are configured")
    public void apiKeysAreConfigured() {
        verifyAPIKeysConfigured();
    }

    @When("I select Stripe as the payment method")
    public void iSelectStripeAsThePaymentMethod() {
        selectStripePaymentMethod();
    }

    @Then("Stripe payment method is selected")
    public void stripePaymentMethodIsSelected() {
        verifyStripePaymentMethodSelected();
    }

    @When("I enter valid credit card details")
    public void iEnterValidCreditCardDetails() {
        enterCreditCardDetails();
    }

    @Then("credit card details are accepted")
    public void creditCardDetailsAreAccepted() {
        verifyCreditCardDetailsAccepted();
    }

    @When("I enter valid billing address")
    public void iEnterValidBillingAddress() {
        enterBillingAddress();
    }

    @Then("billing address is accepted")
    public void billingAddressIsAccepted() {
        verifyBillingAddressAccepted();
    }

    @When("I enter valid email address")
    public void iEnterValidEmailAddress() {
        enterEmailAddress();
    }

    @Then("email address is accepted")
    public void emailAddressIsAccepted() {
        verifyEmailAddressAccepted();
    }

    @When("I enter product ID and amount to be charged")
    public void iEnterProductIDAndAmountToBeCharged() {
        enterProductIDAndAmount();
    }

    @Then("product ID and amount are accepted")
    public void productIDAndAmountAreAccepted() {
        verifyProductIDAndAmountAccepted();
    }

    @When("I submit the payment")
    public void iSubmitThePayment() {
        submitPayment();
    }

    @Then("payment is processed successfully")
    public void paymentIsProcessedSuccessfully() {
        verifyPaymentProcessedSuccessfully();
    }

    @When("I verify transaction confirmation message")
    public void iVerifyTransactionConfirmationMessage() {
        verifyTransactionConfirmationMessage();
    }

    @Then("transaction confirmation message is displayed")
    public void transactionConfirmationMessageIsDisplayed() {
        verifyTransactionConfirmationMessageDisplayed();
    }

    @When("I check Stripe dashboard for transaction record")
    public void iCheckStripeDashboardForTransactionRecord() {
        checkStripeDashboardForTransactionRecord();
    }

    @Then("transaction record is present in Stripe dashboard")
    public void transactionRecordIsPresentInStripeDashboard() {
        verifyTransactionRecordInStripeDashboard();
    }

    @When("I verify email notification for transaction")
    public void iVerifyEmailNotificationForTransaction() {
        verifyEmailNotificationForTransaction();
    }

    @Then("email notification is received")
    public void emailNotificationIsReceived() {
        verifyEmailNotificationReceived();
    }

    @When("I log out from the application")
    public void iLogOutFromTheApplication() {
        logOutFromApplication();
    }

    @Then("user is logged out successfully")
    public void userIsLoggedOutSuccessfully() {
        verifyUserLoggedOutSuccessfully();
    }

    @When("I log back into the application")
    public void iLogBackIntoTheApplication() {
        logBackIntoApplication();
    }

    @Then("user is logged in successfully")
    public void userIsLoggedInSuccessfully() {
        verifyUserLoggedInSuccessfully();
    }

    @When("I check order history for transaction")
    public void iCheckOrderHistoryForTransaction() {
        checkOrderHistoryForTransaction();
    }

    @Then("transaction is present in order history")
    public void transactionIsPresentInOrderHistory() {
        verifyTransactionInOrderHistory();
    }

    @When("I verify customer support is available for transaction queries")
    public void iVerifyCustomerSupportIsAvailableForTransactionQueries() {
        verifyCustomerSupportAvailability();
    }

    @Then("customer support is available")
    public void customerSupportIsAvailable() {
        verifyCustomerSupportAvailable();
    }

    @When("I ensure transaction details are logged for auditing")
    public void iEnsureTransactionDetailsAreLoggedForAuditing() {
        ensureTransactionDetailsLogged();
    }

    @Then("transaction details are logged")
    public void transactionDetailsAreLogged() {
        verifyTransactionDetailsLogged();
    }
}